app/models/dossier.rb

```diff
@@ -939,4 +942,17 @@ def self.notify_draft_not_submitted
         DossierMailer.notify_brouillon_not_submitted(dossier).deliver_later
       end
   end
+
+  def send_dossier_decision_to_experts(dossier)
+    experts = Avis.joins(:experts_procedure)
+      .where(dossier: dossier, experts_procedures: { allow_decision_access: true })
+      .where.not(answer: nil)
+      .map(&:expert)
```

Comment:
- on n'a pas besoin de fournir dossier en parametre et on peut directement appeler la relation `avis`
- on utilise le scope `with_answer`
- on ne veut qu'un seul retour par expert mm si celui-ci a donné plusieurs avis : `distinct`
- on n'a pas besoin de tout l'objet, juste de l'avis.id mais on n'en veut qu'un seul par experts_procedure pour éviter d'envoyer plusieurs mails a un mm expert

```suggestion
    avis_experts_procedures_ids = avis
      .joins(:experts_procedure)
      .with_answer
      .where(experts_procedures: { allow_decision_access: true })
      .distinct
      .pluck('avis.id, experts_procedures.id')
      
      # select only one avis per experts_procedures
      avis_ids = avis_experts_procedures_ids
        .uniq { |(avis_id, experts_procedures_id)| experts_procedures_id }
        .map{ |(avis_id, _)| avis_id }
```

et ca serait pas mal de tester, a minima que ca selectionne seulement un avis par expert.